svr-main.c

```diff
@@ -356,6 +356,19 @@ static void main_noinetd() {
 }
 #endif /* NON_INETD_MODE */
 
+#if DROPBEAR_RESTRICT_FIXED_CLIENT_IP
+static int validate_client_ip(const char* remote_host, const char* allowed_client_ip_addr)
+{
+	return (allowed_client_ip_addr != NULL) && (strcmp(remote_host, allowed_client_ip_addr) != 0);
```

Comment:
isn't it a global variable?
The logic is wrong, it should be:
allowed != NULL => they match

Reply:
is NULL || they match

Reply:
Done.